edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.CloudProxy.Test/MessageHelper.cs

```diff
@@ -51,18 +51,26 @@ public static IList<IMessage> GenerateMessages(int count)
         /// Checks if the events have items corresponding to all messages, and
         /// returns false if it doesn't.
         /// </summary>
-        public static bool CompareMessagesAndEventData(IList<IMessage> messages, IList<EventData> events)
+        public static bool ValidateSentMessagesWereReceived(IDictionary<string, IList<IMessage>> sentMessagesByDevice, IDictionary<string, List<EventData>> receivedMessagesByDevice)
         {
-            foreach (IMessage message in messages)
+            foreach (string deviceId in sentMessagesByDevice.Keys)
```

Comment:
I think you need to check number of keys in receivedMessagesByDevice should be greater than or equal to number of keys in sentMessagesByDevice. Similar check you can do before verifying each message, check receivedMessagesByDevice[deviceId].Count should be greater than or equal to sentMessagesByDevice[deviceId].Count.

btw receivedMessagesByDevice doesn't mean it stores received message only for that device.  It stores received messages only for that partition id derived by that device id.  do i understand correctly? if so, please rename it.

Reply:
Isn't the check I'm doing already sufficient? The current logic makes sure sent messages is a subset of the received messages for each device.

You understand correctly. I will rename.